src/app/market/trades/trades.component.ts

```diff
@@ -0,0 +1,93 @@
+import _ from 'lodash';
+import moment from 'moment';
+import Promise from 'bluebird';
+import {Component} from '@angular/core';
+
+import {MarketDAO} from '../../shared/DAO/market.dao';
+import {ToastService} from '../../shared/services/toast.service';
+
+import t from '../../shared/defineTextToTranslate';
+
+@Component({
+  selector: 'app-market-trades',
+  templateUrl: 'trades.component.html'
+})
+export class TradesComponent {
+
+  public data: any;
+  public currencyFilter: string;
+  public interval: moment.unitOfTime.DurationConstructor;
+  public supportedPaymentsCurrencies: Array<Object>;
+  public supportedIntervals: Array<Object>;
+  public noData: boolean = false;
+  public loading: boolean = false;
+  public trades: Array<Object>;
+
+  constructor(private marketDAO: MarketDAO,
+              private toast: ToastService) {
+    this.supportedIntervals = [
+      {
+        label: t('MARKET.TRADE.INTERVAL.DAY'),
+        value: 'day'
+      },
+      {
+        label: t('MARKET.TRADE.INTERVAL.WEEK'),
+        value: 'week'
+      },
+      {
+        label: t('MARKET.TRADE.INTERVAL.MONTH'),
+        value: 'month'
+      },
+      {
+        label: t('MARKET.TRADE.INTERVAL.YEAR'),
+        value: 'year'
+      },
+    ];
+  }
+
+  ngOnInit() {
+    this.marketDAO.getAvailableMarkets().then((res: any) => {
+      this.supportedPaymentsCurrencies = _.chain(res).filter(o => o.lsymbol === 'BTC' && o.rtype === 'fiat').value();
+    });
+    this.interval = 'day';
+  }
+
+  private parseNumber(d) {
```

Comment:
I don't understand what this method is doing by reading this function name and code.

Reply:
is the name `parseNumberValuesInObject` good?

Reply:
This approach is strange, why do you want to convert every single value?

Reply:
OK..  I parse only known keys.